File: onyx-remote-driver/src/main/java/com/onyx/client/base/RequestToken.java
package com.onyx.client.base;

import com.onyx.buffer.BufferStream;
import com.onyx.buffer.BufferStreamable;
import com.onyx.exception.BufferingException;
import com.onyx.persistence.context.SchemaContext;

import java.io.Serializable;

/**
 * Created by tosborn1 on 2/10/17.
 *
 * This class is a token that is sent back and fourth between the client and server.
 */
public class RequestToken implements BufferStreamable
{
    public boolean reTry;
    public short token;
    public Serializable packet;

    public RequestToken()
    {

    }

    public RequestToken(short token, Serializable packet)
    {
        this.token = token;
        this.packet = packet;
    }

    @Override
    public int hashCode()
    {
        return token;
    }

    @Override
    public boolean equals(Object o)
    {
        return (o != null && o instanceof RequestToken && ((RequestToken) o).token == token);
    }

    @Override
    public void read(BufferStream buffer) throws BufferingException {
        token = buffer.getShort();
        packet = (Serializable)buffer.getObject();
    }

    @Override
    public void write(BufferStream buffer) throws BufferingException {
        buffer.putShort(token);
        buffer.putObject(packet);
    }

    @Override
    public void read(BufferStream bufferStream, SchemaContext context) throws BufferingException {
        read(bufferStream);
    }

    @Override
    public void write(BufferStream bufferStream, SchemaContext context) throws BufferingException {
        write(bufferStream);
    }

}

File: onyx-remote-driver/src/main/java/com/onyx/client/base/RequestToken.kt
package com.onyx.client.base

import com.onyx.buffer.BufferStream
import com.onyx.buffer.BufferStreamable
import com.onyx.exception.BufferingException

import java.io.Serializable

/**
 * Created by tosborn1 on 2/10/17.
 *
 * This class is a token that is sent back and fourth between the client and server.
 */
class RequestToken() : BufferStreamable {

    var token: Short = 0
    var packet: Any? = null

    constructor(token: Short, packet: Any?): this() {
        this.token = token
        this.packet = packet
    }

    @Throws(BufferingException::class)
    override fun read(buffer: BufferStream) {
        token = buffer.short
        packet = buffer.value as Serializable?
    }

    @Throws(BufferingException::class)
    override fun write(buffer: BufferStream) {
        buffer.putShort(token)
        buffer.putObject(packet)
    }

    override fun hashCode(): Int = token.toInt()

    override fun equals(other: Any?): Boolean = other != null && other is RequestToken && other.token == token


}